File: 0929-unique-email-addresses/0929-unique-email-addresses.java
class Solution {
    public int numUniqueEmails(String[] emails) {
        HashSet<String> set = new HashSet<>();
        for(String e : emails){
            String[] t = e.split("@");
            String x = fix(t[0]) + "@" + t[1];
            set.add(x);
        }
        return set.size();
    }
    public String fix(String s){
        StringBuilder sb = new StringBuilder();
        for(int i = 0 ; i < s.length() && s.charAt(i) != '+' ; i++){
            if(s.charAt(i) != '.') sb.append(s.charAt(i));
        }
        return sb.toString();
    }
}

File: 0929-unique-email-addresses/0929-unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        hashset = set()
        for email in emails:
            email = self.process_email(email)
            hashset.add(email)
        return len(hashset)
    
    def process_email(self, email: str) -> str:
        local, domain = email.split('@')
        local_plus = local.split('+')[0].replace('.', '')
        return f'{local_plus}@{domain}'
        